### ISSUE TITLE
Does this task work on non-windows agents?

### ISSUE THREAD

**Issue by btrepp** [renevanosnabrugge/vsts-promotepackage-task#57]
Not 100% sure if non-windows OS is supported?

Running through azure devops yaml piplines, with 'ubuntu-latest'.
Gives the below.

![image](https://user-images.githubusercontent.com/262933/83599549-0cb82800-a59f-11ea-8b88-835662cfd290.png)

**Comment by jessehouwing** [comment]
It does not. The task is built with Powershell and the Linux and Mac agents
don't support tasks written in powershell as long as Microsoft doesn't add
a powershell core executionhandler.

You could add a 2nd job to your pipeline, have it run Windows and do the
promotion from there as a workaround.

**Comment by DeluxZ** [comment]
@jessehouwing  Maybe we can add support for PowerShell Core? You can run the 'normal' PowerShell@2 task with the input `pwsh: true`.

**Comment by jessehouwing** [comment]
> @jessehouwing  Maybe we can add support for PowerShell Core? You can run the 'normal' PowerShell@2 task with the input `pwsh: true`.

No, it's not easy to convert this extension to pwsh-core. The agent initializes the context for the task and for the SDK to work. The agent has 3 modes:

 * powershell runtime hosted. The agent will actually host the powershell code. This mode is deprecated since October 2019.
 * powershell3 is an out of process host which spins up a powershell instance and has it execute the task.
 * Node/Node10 is similar to powershell but fires up a Node 6 or Node 10 runtime to execute the task.

Only the Node/Node10 runtime is supported on non-windows-agents. So to enable this task on Linux, the task needs to be ported to a task host that is cross-platform. Maybe a thin wrapper in Node could in turn invoke pwsh, but that introduces a bunch of new headaches around return codes, error stream handling and such.
